http_logging.go

```diff
@@ -74,6 +75,96 @@ func LogRequest(r *http.Request, msg string, extraAttributes ...KeyVal) {
 	S(Info, msg, attr...)
 }
 
+// LogResponse logs the response code, byte size and duration of the request.
+// additional key:value pairs can be passed as extraAttributes.
+//
+//nolint:revive // name is fine.
+func LogResponse[T *ResponseRecorder | *http.Response](r T, msg string, extraAttributes ...KeyVal) {
+	if !Log(Info) {
+		return
+	}
+	var status int
+	var size int64
+	switch v := any(r).(type) { // go generics...
+	case *ResponseRecorder:
+		status = v.StatusCode
+		size = v.ContentLength
+	case *http.Response:
+		status = v.StatusCode
+		size = v.ContentLength
+	}
+	attr := []KeyVal{
+		Int("status", status),
+		Int64("size", size),
+	}
+	attr = append(attr, extraAttributes...)
+	S(Info, msg, attr...)
+}
+
+// Can be used (and is used by LogAndCall()) to wrap a http.ResponseWriter to record status code and size.
+type ResponseRecorder struct {
+	w             http.ResponseWriter
+	startTime     time.Time
+	StatusCode    int
+	ContentLength int64
+}
+
+func (rr *ResponseRecorder) Header() http.Header {
+	return rr.w.Header()
+}
+
+func (rr *ResponseRecorder) Write(p []byte) (int, error) {
+	size, err := rr.w.Write(p)
+	rr.ContentLength += int64(size)
+	if err != nil {
+		rr.StatusCode = http.StatusInternalServerError
+	} else if rr.StatusCode == 0 {
+		rr.StatusCode = http.StatusOK
+	}
+	return size, err
+}
+
+func (rr *ResponseRecorder) WriteHeader(code int) {
+	rr.w.WriteHeader(code)
+	rr.StatusCode = code
+}
+
+// LogAndCall logs the incoming request and the response code, byte size and duration
+// of the request.
+//
+// If Config.CombineRequestAndResponse or the LOGGER_COMBINE_REQUEST_AND_RESPONSE
+// environment variable is true. then a single log entry is done combining request and
+// response information, including catching for panic.
+//
+// Additional key:value pairs can be passed as extraAttributes.
+//
+//nolint:revive // name is fine.
+func LogAndCall(msg string, hf http.HandlerFunc, extraAttributes ...KeyVal) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if Config.CombineRequestAndResponse {
+			rr := &ResponseRecorder{w: w, startTime: time.Now()}
+			defer func() {
+				if err := recover(); err != nil {
+					S(Critical, "panic in handler", Any("error", err))
+				}
+				attr := []KeyVal{
+					Int("status", rr.StatusCode),
+					Int64("size", rr.ContentLength),
+					Int64("microsec", time.Since(rr.startTime).Microseconds()),
+				}
+				attr = append(attr, extraAttributes...)
+				LogRequest(r, msg, attr...)
+			}()
+			hf(rr, r)
+			return
+		}
+		LogRequest(r, msg, extraAttributes...)
+		rr := &ResponseRecorder{w: w, startTime: time.Now()}
+		hf(rr, r)
```

Comment:
Just personal nit/fyi (no need to change) but I think out of context `hf(rr,r)` makes the code **reading** quite complex. I don't know why Go leans towards this type of behaviour but I _really_ like clear variable names myself. Obviously it's understandable by moving up and/or be more involved in the code-base. 

Somewhat the same applies to this func: https://github.com/fortio/log/pull/53/files#diff-5753fc63f23c85fb673bc9c18b757335e8ddab9e4ab582ad2194eb1544992ab9R74-R133

Reply:
changing this to improve readability (hopefully it improves enough that the test can remain somewhat terse)

Reply:
ptal https://github.com/fortio/log/pull/53/commits/a3785bef24561594dc154af560ebb9ec411aa311

Reply:
I like it

Reply:
thx will merge and propagate tomorrow (to scli, dflag, fortio... maybe I should go back to mono repo :) )